api/handlers/process.go

```diff
@@ -45,19 +48,25 @@ type Process struct {
 	processRepo      CFProcessRepository
 	processStats     ProcessStats
 	requestValidator RequestValidator
+	podRepo          PodRepository
+	appRepo          CFAppRepository
 }
 
 func NewProcess(
 	serverURL url.URL,
 	processRepo CFProcessRepository,
 	processStatsFetcher ProcessStats,
 	requestValidator RequestValidator,
+	appRepo CFAppRepository,
+	podRepo PodRepository,
 ) *Process {
 	return &Process{
 		serverURL:        serverURL,
 		processRepo:      processRepo,
 		processStats:     processStatsFetcher,
 		requestValidator: requestValidator,
+		appRepo:          appRepo,
```

Comment:
Injecting an app repo to the process handler looks weird to me. We only need the app record so that we know the app revision, so maybe we could extend the process repo by adding a `GetAppRevisionForProcess` method. It would not duplicate a lot of the app repo code since it is simply a k8sClient.Get call. On the other hand the app repo contains a lot of methods that the process repo should not be able to see.